src/modules/rbd/rbd-list/rbd-list.js

```diff
@@ -83,6 +92,47 @@
         function createRbd() {
             $state.go("create-rbd");
         }
+
+        function rbdResizeModal(rbdObject) {
+            vm.resizeRBDtaskSubmitted = false;
+            var used, sizeAndUnit = [], clusterObj;
+            vm.resizeRbd = rbdObject;
+            if(rbdObject.used) {
+                used = $filter('bytes')(rbdObject.used);
+                sizeAndUnit = used.split(' ');
+                vm.resizeRbd.size = Math.round(parseFloat(sizeAndUnit[0]));
+                vm.resizeRbd.unit = sizeAndUnit[1];
+            }
+            clusterObj = utils.getClusterDetails(rbdObject.clusterId);
+            console.log("this is the cluster obj " , clusterObj);
+            vm.resizeRbd.clusterAvailable = clusterObj.utilization.available;
+        }
+
+
+        function resizeRBD() {
+            var sizeInBytes, sizeInMB, postData;
+            console.log("this is the unit " , vm.resizeRbd.unit );
+            sizeInBytes = utils.convertToBytes(vm.resizeRbd.size, vm.resizeRbd.unit);
+            sizeInMB = (sizeInBytes / (1024*1024)).toFixed(0);
+            vm.resizeRbd.size = parseInt(sizeInMB);
+
+            postData = { "Rbd.pool_id":vm.resizeRbd.pool_id, "Rbd.name": vm.resizeRbd.name, "Rbd.size": vm.resizeRbd.size };
+            
+            utils.takeAction(postData, "CephResizeRbd", "PUT", vm.resizeRbd.clusterId).then(function(response) {
+                $rootScope.notification.type = "success";
+                $rootScope.notification.message = "JOB is under process. and JOB-ID is - " + response.job_id;
+            });
+
+            vm.resizeRBDtaskSubmitted = true;
+        }
+
+
+        function viewTaskProgress() {
+            angular.element('#rbdResizeModal').modal('hide');
```

Comment:
@kamleshverma1 You can do this by using flag in resize-rbd.hmtl for eg ng-if="rbdlCntrl.resizetButtonClicked instead of using hide and display in controller.